sway/desktop/layer_shell.c

```diff
@@ -139,18 +143,12 @@ static void arrange_layer(struct sway_output *output, struct wl_list *list,
 			box.y = bounds.y + ((bounds.height / 2) - (box.height / 2));
 		}
 		// Margin
-		if ((state->anchor & both_horiz) == both_horiz) {
-			box.x += state->margin.left;
-			box.width -= state->margin.left + state->margin.right;
```

Comment:
If the client requests 0 width and opposing anchors we still need to honor the margin, so we can't drop this.

I'm thinking the cases should really be "if width == 0" and "if left else if right". wlroots guarantees width == 0 implies both horizontal anchors by enforcing the protocol.

Reply:
Oh right, width = 0 means as wide as possible. I re added it and changed the check to `width = 0` (moving the width changing below).